Python/Product/Debugger/DebugAdapterUtils.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Python Tools for Visual Studio
+// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using Microsoft.Win32;
+using System;
+using System.Diagnostics;
+
+namespace Microsoft.PythonTools {
+    public class DebugAdapterUtils {
+        public static void GetProcessInfoFromUri(Uri uri, out int pid, out string processName) {
+            ProcessStartInfo psi = new ProcessStartInfo {
+                FileName = @"C:\Windows\System32\NETSTAT.EXE",
+                Arguments = "-a -n -o -p TCP",
+                UseShellExecute = false,
+                RedirectStandardOutput = true,
+                RedirectStandardInput = true,
+                RedirectStandardError = true
+            };
+
+            using (var process = Process.Start(psi)) {
```

Comment:
This needs a check that `uri` is loopback, first, otherwise we might get an accidental match.

Also, `Process.Start` can throw a bunch of stuff; not sure about `WaitForExit`. But basically this needs to gracefully handle any errors during retrieval - we don't want to fail to show the process as available to attach just because we couldn't get info on it.

(If you feel like dropping this workaround instead, and just going with the hardcoded PID and name, that works too. Showing accurate information for remote processes is not a goal at this stage. The important part is that the endpoint gets validated, and can be connected to.

Reply:
I agree, will remove this workaround.

Reply:
We had some good implementation of this in RTVS.

Reply:
I think it is risky to do this using netstat directly. There are other issues involving user permissions with using netstat. If needed i will implement this using the underlying API used by netstat to avoid process handle probing which requires additional privileges.